sigstore-java/src/main/java/dev/sigstore/trustroot/TransparencyLog.java

```diff
@@ -59,4 +58,21 @@ static Optional<TransparencyLog> find(List<TransparencyLog> all, byte[] logId, I
         .filter(tl -> tl.getPublicKey().getValidFor().contains(time))
         .findAny();
   }
+
+  /**
+   * Find a log by list of validity times and logId. This will find the first log with matching log
+   * id that was valid at all times.
+   *
+   * @param logId the logId of the log
+   * @param times the list of timea the log was expected to be valid
+   * @return the first log that was valid at all {@code times} with logId {@code logId}
+   */
+  static Optional<TransparencyLog> find(
```

Comment:
we run the risk of timestamps that don't really matter affecting how we find a log here and I think we might need to consider the implications of that.